engine/api/src/main/java/com/cloud/vm/VirtualMachineManager.java

```diff
@@ -255,4 +256,10 @@ static String getHypervisorHostname(String name) {
     boolean unmanage(String vmUuid);
 
     UserVm restoreVirtualMachine(long vmId, Long newTemplateId) throws ResourceUnavailableException, InsufficientCapacityException;
+
+    Pair<Long, Long> findClusterAndHostIdForVmFromVolumes(long vmId);
+
+    Pair<Long, Long> findClusterAndHostIdForVm(VirtualMachine vm);
+
+    Pair<Long, Long> findClusterAndHostIdForVm(long vmId);
```

Comment:
As methods are related, why not leaving only this one and make the first two methods as private on the implementation?

Reply:
Made other methods private